osu.Game.Rulesets.Osu/Objects/Drawables/DrawableHitCircle.cs

```diff
@@ -264,12 +259,31 @@ public HitReceptor()
 
             public bool OnPressed(KeyBindingPressEvent<OsuAction> e)
             {
+                if (!(CanBeHit?.Invoke() ?? false))
```

Comment:
Good lord this is convoluted. I have to write down a truth table to even figure out what this is trying to compute.

Can we simplify to

```suggestion
                if (CanBeHit?.Invoke() != true)
```

(or at least I *think* that's the equivalent...)

Reply:
I've taken a bit of a different path, by enabling NRT for this file.

In the process, I've taken liberty of using a little bit of a newer syntax: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/required

I hope this is okay and agreeable (I think it can be used in more places). The alternative would be to use the ctor to indicate that it's required, but imo this is a _teeny_ bit cleaner.

Reply:
I could be fine with `required` properties as long as @peppy is. I'm personally a fan.